component/client/chat/ClientChat.tsx

```diff
@@ -48,15 +49,15 @@ class ClientChat extends React.Component<IClientChatProps, IClientChatState> {
       messages : []
     };
     room.setUrl(url, (data: IData, user) => {
-      this.setState((prevState, props) => {
+      this.setState((prevState: IClientChatState, props: IClientChatProps) => {
```

Comment:
I don't think it is necessary to have fromUserID in IData.
The second argument user here was intended to be the userID.
I did this refactoring in [The roomService pull request](https://github.com/zhoutwo/url_connect/pull/28)